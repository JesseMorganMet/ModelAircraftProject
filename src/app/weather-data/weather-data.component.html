<h1>{{data.location.name}}</h1>
<p><b>Wind speed: </b> {{data.timeSeries[0].max10mWindGust}}</p>
<p><b>Chance of rain: </b> {{data.timeSeries[0].probOfPrecipitation}}</p>
<p><b>Visibility: </b> {{data.timeSeries[0].visibility}}</p>
