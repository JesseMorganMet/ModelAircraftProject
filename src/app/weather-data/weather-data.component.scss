h1 {
  background-color: #3b3b3b;
  box-sizing: border-box;
  width: 100%;
  color: #fff;
  padding: 1%;
  margin: 0;
}

p {
  background-color: white;
  padding: 1%;
  margin: 0 2% 0 2%;
}

@media (max-width: 800px) {
  h1 {
    font-size: 1em;
  }
  p {
    font-size: 0.8em;
  }
}
