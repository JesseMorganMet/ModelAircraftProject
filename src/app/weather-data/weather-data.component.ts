import {Component, Input, OnInit} from '@angular/core';

@Component({
  selector: 'app-weather-data',
  templateUrl: './weather-data.component.html',
  styleUrls: ['./weather-data.component.scss']
})
export class WeatherDataComponent implements OnInit {
  @Input() dataLocation: any;
  @Input() dataWithTime: any;
  @Input() weatherParameters: any;

  i: number = 0;

  checker: any;
  checkAttributes: any;

  constructor() {
  }

  ngOnInit() {
    this.masterFunction();
  }

  ngOnChanges(change) {
    console.log(change)
    this.masterFunction();
  }

  masterFunction() {
    this.checkAttributes = [];
    this.windChecker(this.dataWithTime.windSpeed10m);
    this.precipitationChecker(this.dataWithTime.probOfPrecipitation);
    this.visibilityChecker(this.dataWithTime.visibility);
    this.optimalWeather();
  }

  windChecker(data) {
    if (data <= this.weatherParameters.wind) {
      this.checkAttributes.push({optimal: true, class: "optimal", text: " Optimal"})
      return this.checkAttributes;
    } else {
      this.checkAttributes.push({optimal: false, class: "notOptimal", text: " Not Optimal"})
      return this.checkAttributes;
    }
  }

  precipitationChecker(data) {
    if (data <= this.weatherParameters.rain) {
      this.checkAttributes.push({optimal: true, class: "optimal", text: " Optimal"})
      return this.checkAttributes;
    } else {
      this.checkAttributes.push({optimal: false, class: "notOptimal", text: " Not Optimal"})
      return this.checkAttributes;
    }
  }

  visibilityChecker(data) {
    if (data >= this.weatherParameters.vis) {
      this.checkAttributes.push({optimal: true, class: "optimal", text: " Optimal"})
      return this.checkAttributes;
    } else {
      this.checkAttributes.push({optimal: false, class: "notOptimal", text: " Not Optimal"})
      return this.checkAttributes;
    }
  }

  optimalWeather() {
    for (let bools of this.checkAttributes) {
      if (bools.optimal == false) {
        this.checker = "Weather Requirements Are Not Optimal, NO GO";
        break;
      } else if (bools.optimal == true) {
        this.checker = "Weather is Optimal, GO";
      } else {
        this.checker = "Something went wrong";
      }
    }
  }
}
