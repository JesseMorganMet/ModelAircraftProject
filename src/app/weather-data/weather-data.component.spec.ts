import {ComponentFixture, TestBed} from '@angular/core/testing';

import {WeatherDataComponent} from './weather-data.component';
import {pOPrecipPipe, visiblityPipe, WindSpeedPipe} from '../weather-data.pipe';

describe('WeatherDataComponent', () => {
  let component: WeatherDataComponent;
  let fixture: ComponentFixture<WeatherDataComponent>;
  const mockDataL = {
    "name": "mike"
  };
  const mockDataT = {
    "time": "2022-08-26T13:00Z",
    "max10mWindGust": 10,
    "probOfPrecipitation": 5,
    "visibility": 20
  }


  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        WeatherDataComponent,
        WindSpeedPipe,
        pOPrecipPipe,
        visiblityPipe
      ]
    })
      .compileComponents();

    fixture = TestBed.createComponent(WeatherDataComponent);
    component = fixture.componentInstance;
  });

  it('Should match snapshot', () => {
    component.dataLocation = mockDataL;
    component.dataWithTime = mockDataT;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  describe('Should test windChecker function', () => {
    it('Should return True', () => {
      component.checkAttributes = [];
      let data = 2;
      component.windSpeedSuitability = 5;
      component.windChecker(data);
      expect(component.checkAttributes[0].optimal).toBe(true);
    });
    it('Should Return False', () => {
      component.checkAttributes = [];
      let data = 10;
      component.windSpeedSuitability = 5;
      component.windChecker(data);
      expect(component.checkAttributes[0].optimal).toBe(false);
    });
  });

  describe('Should test precipitationChecker function', () => {
    it('Should return True', () => {
      component.checkAttributes = [];
      let data = 2;
      component.precipitationSuitability = 5;
      component.precipitationChecker(data);
      expect(component.checkAttributes[0].optimal).toBe(true);
    });
    it('Should Return False', () => {
      component.checkAttributes = [];
      let data = 10;
      component.precipitationSuitability = 5;
      component.precipitationChecker(data);
      expect(component.checkAttributes[0].optimal).toBe(false);
    });
  });

  describe('Should test visibilityChecker function', () => {
    it('Should return True', () => {
      component.checkAttributes = [];
      let data = 10;
      component.visibilitySuitability = 5;
      component.visibilityChecker(data);
      expect(component.checkAttributes[0].optimal).toBe(true);
    });
    it('Should Return False', () => {
      component.checkAttributes = [];
      let data = 2;
      component.visibilitySuitability = 5;
      component.visibilityChecker(data);
      expect(component.checkAttributes[0].optimal).toBe(false);
    });
  });
  describe('Should test optimalWeather function', () => {
    it('Should return String if anything with key of optimal is false', () => {
      component.checkAttributes = [{optimal:true},{optimal:true},{optimal:false}]
      component.optimalWeather();
      expect(component.checker).toBe("Weather Requirements Are Not Optimal, NO GO");
    });
    it('Should Return String if everything with key of optimal is True', () => {
      component.checkAttributes = [{optimal:true},{optimal:true},{optimal:true}]
      component.optimalWeather();
      expect(component.checker).toBe("Weather is Optimal, GO");
    });
    it('Should Return String if neither', () => {
      component.checkAttributes = [{},{},{}]
      component.optimalWeather();
      expect(component.checker).toBe("Something went wrong");
    });
  });

});
