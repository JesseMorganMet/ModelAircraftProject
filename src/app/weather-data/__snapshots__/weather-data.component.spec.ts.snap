// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDataComponent Should match snapshot 1`] = `
<app-weather-data
  __ngContext__="0"
  checkAttributes={[Function Array]}
  checker={[Function String]}
  dataLocation={[Function Object]}
  dataWithTime={[Function Object]}
  i="0"
  weatherParameters={[Function Object]}
>
  <h1>
    mike
    <br />
    (8/26/22, 2:00 PM)
  </h1><h2>
    Weather is Optimal, GO
  </h2><div
    class="weatherInfo"
  >
    <p>
       Wind speed: 
      <span
        class="dataValue"
      >
        10m/s
      </span>
      <span
        class="optimal"
      >
         Optimal
      </span>
    </p>
    <p>
       Chance of rain: 
      <span
        class="dataValue"
      >
        5%
      </span>
      <span
        class="optimal"
      >
         Optimal
      </span>
    </p>
    <p>
       Visibility: 
      <span
        class="dataValue"
      >
        Very Poor (20m)
      </span>
      <span
        class="optimal"
      >
         Optimal
      </span>
    </p>
  </div>
</app-weather-data>
`;
